### ISSUE TITLE
Preview feature feedback: PostgreSQL extensions

### ISSUE THREAD

**Issue by floelhoeffel** [prisma/prisma#15835]
Please use this issue to share any **feedback** about the `postgresqlExtensions` functionality released in [v4.5.0](https://github.com/prisma/prisma/releases/tag/4.5.0).

**Documentation** can be found [here](https://prisma.io/concepts/components/prisma-schema/postgresql-extensions).

If you encounter a **bug**, please [open a bug report](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=bug_report.md&title=) in this repo.

Many thanks 👏🏻 from the Prisma team.

**Comment by chriskuech** [comment]
Please add support for `cube`

**Comment by chriskuech** [comment]
@Jolg42 Thanks. Doesn't that mean you need to add `cube` to the "allowlist" here?: 
<img width="720" alt="image" src="https://github.com/prisma/prisma/assets/3680671/187eea6f-f8b4-4d3f-a9b1-5f88bdba9ba8">

https://www.prisma.io/docs/orm/prisma-schema/postgresql-extensions

**Comment by Jolg42** [comment]
@chriskuech So maybe this is confusing because it's more of a "well known list", what happens is that if an extension is on this list then the Introspection behavior changes, see below:
<img width="550" alt="Screenshot 2024-01-30 at 10 54 25" src="https://github.com/prisma/prisma/assets/1328733/904b3012-fc85-41c2-a863-0342c9bf1045">

Any extension can be used actually, the only difference will be in the behavior of Introspection / Re-Introspection.

I think I remember something about some extensions being safe to add or remove during migrations, but that's not guaranteed for any extension, so to be on the safe side we have this list.

**Comment by foxfriends** [comment]
I have an extension installed for development purposes only (specifically: pgtap is manually installed on my local instance for testing, but intentionally not installed on my deployed database).

Currently, each time I try to migrate after installing pgtap to my local database and running my tests, Prisma tries to reset. This happens whether I have `pgtap` listed in my schema or not, because it detects `pgtap` being installed but not having been created by any migration.

Seeing as this extension will have no affect on Prisma or my application, whether installed or not, it would be useful for to be a way to mark an extension as "optional" or "development only" or "ignored" or something, so that it just... doesn't get involved at any time. Similar to the existing `@ignore` directive for fields/tables.

**Comment by Aluisio** [comment]
@floelhoeffel When will this feature be promoted to general availability?

**Comment by why-el** [comment]
I agree with @foxfriends, this really needs to allow excluding some existing extensions. I attempted to adopt it for my team, but since we use cloud hosted databases even for developers, we cannot use this feature unless it allows us to exclude extensions that exist in the cloud instances but we do not care for them that much. If we can't ignore them, they cause drifts with the shadow database and each Prisma operation wants to delete everything. 

The Prisma team is allowlisting a few extensions. I think that's the right approach but should be provided as a utility to the user as well. There is no way you guys have the time (or the energy) to find and allowlist every approved extension from every cloud provider under the sun 😆.

**Comment by sprutner** [comment]
I have the same issue with @why-el and @foxfriends

**Comment by sprutner** [comment]
> @sprutner, @why-el and @foxfriends - would it work for you if we added the ability to explicitly disable specific extensions? Perhaps with a syntax like this to specifically disable pgtab:
> 
> 
> 
> ```
> 
> datasource db {
> 
>   provider   = "postgresql"
> 
>   url        = env("DATABASE_URL")
> 
>   extensions = [postgis(version: "2.1"), pgtap(enabled: false)]
> 
> }
> 
> ```
> 
> 
> 
> And with support for environment variables like this:
> 
> 
> 
> ```
> 
> datasource db {
> 
>   provider   = "postgresql"
> 
>   url        = env("DATABASE_URL")
> 
>   extensions = [postgis(version: "2.1"), pgtap(enabled: env("IS_DEVELOPMENT"))]
> 
> }
> 
> ```
> 
> 
> 
> The `enabled` argument will default to true, so it only changes the behaviour when explicitly set to false.

Your two examples are identical. I needed a way to disable as it was failing migrations. I can't remember how I worked around it that was awhile ago. Being able to disable would be useful as sometimes there's a race condition in running the migration to disable it fails due to it being enabled.

**Comment by wmadden** [comment]
**Update: Discontinuation planned for Jun - Aug 2025**

After careful evaluation, we’ve determined that the current implementation of the postgresqlExtensions preview feature is too complex and doesn’t provide enough value.

While this specific feature will be discontinued, we remain committed to supporting Postgres extensions. Instead of this approach, we’ll ship support for specific extensions—such as PostGIS—through focused, independent efforts.

Read more here: https://github.com/prisma/prisma/discussions/26136

**Comment by Aluisio** [comment]
I currently use
```
generator client {
  provider = "prisma-client-js"
  previewFeatures = ["postgresqlExtensions"]
}

datasource db {
  provider = "postgresql"
  url = env("DATABASE_URL")
  extensions = [citext]
}
```

to use
```
  model User {
  id String   @id @default(uuid())
  name        String
  email        String @db.Citext
  password String
}
```
Does this mean that from June - August I will no longer be able to use this approach? Or will there still be support for citext?

**Comment by JClackett** [comment]
Im relying on pg_vector in production + prisma, what will this mean I have to do to keep this working?

**Comment by tats-u** [comment]
Japanese virtually requires a PostgreSQL extension [pg_bigm](https://github.com/pgbigm/pg_bigm) or [PGroonga](https://pgroonga.github.io/) to perform a full text search if you don't rely on Elasticsearch or Meilisearch.
PGroonga is available in [Supabase](https://supabase.com/docs/guides/database/extensions/pgroonga), and pg_bigm is available in [Aurora](https://aws.amazon.com/about-aws/whats-new/2021/06/amazon-aurora-postgresql-supports-pg-bigm-extension-for-faster-full-text-search/) and [Cloud SQL](https://cloud.google.com/sql/docs/postgres/extensions).
Both extensions of course support other languages.
